### DISCUSSION TITLE
Merging custom latin font with Unicode font problem.

### DISCUSSION THREAD

**Discussion by ynd3fined** [ocornut/imgui#5949] [New Users Build/Link/Run issues ONLY!]
I want merge the "Arial Unicode MS" font with my custom fonts because when i push my custom font with `PushFont`, shows alphabets as question marks. I want to make the texts as in the picture below. But when i make this, application uses high ram and when i add more than two font, application starts but imgui not showing. (I'm using "Arial Unicode MS" font to show most alphabets)

![image](https://user-images.githubusercontent.com/112515666/205445824-d9b6b67c-b32d-4a42-8b56-9a20d5a59963.png)

I don't add latin characters range to "Arial Unicode MS" font because when I push custom font, showing "Arial Unicode MS" characters therefore i removed all latin ranges in `GetGlyphRanges...` functions.

My codes:
```c++
  static const ImWchar arabicchar_ranges[] =
  {
      0x2010, 0x205E, // Punctuations
      0x0600, 0x06FF, // Farsi/Arabic
      0xFE00, 0xFEFF, // Farsi/Arabic
      0,
  };

  ImVector<ImWchar> ranges;
  ImFontGlyphRangesBuilder builder;
  builder.AddRanges(io.Fonts->GetGlyphRangesJapanese());
  builder.AddRanges(io.Fonts->GetGlyphRangesChineseFull());
  builder.AddRanges(io.Fonts->GetGlyphRangesChineseSimplifiedCommon());
  builder.AddRanges(io.Fonts->GetGlyphRangesCyrillic());
  builder.AddRanges(io.Fonts->GetGlyphRangesKorean());
  builder.AddRanges(io.Fonts->GetGlyphRangesThai());
  builder.AddRanges(io.Fonts->GetGlyphRangesVietnamese());
  builder.AddRanges(arabicchar_ranges);
  builder.BuildRanges(&ranges);

  //This range supported by my custom font
  static const ImWchar latinsupchar_ranges[] =
  {
      0x0020, 0x00FF, // Basic Latin + Latin Supplement
      0x00c7, 0x00c7, // Ç
      0x00e7, 0x00e7, // ç
      0x011e, 0x011e, // Ð
      0x011f, 0x011f, // ð
      0x0130, 0x0130, // Ý
      0x0131, 0x0131, // ý
      0x00d6, 0x00d6, // Ö
      0x00f6, 0x00f6, // ö
      0x015e, 0x015e, // Þ
      0x015f, 0x015f, // þ
      0x00dc, 0x00dc, // Ü
      0x00fc, 0x00fc, // ü
      0
  };

  //Merge Config
  ImFontConfig config;
  config.MergeMode = true;
  config.GlyphRanges = latinsupchar_ranges;
  
  //Custom Font 1
  io.Fonts->AddFontFromFileTTF("Arial Unicode MS Font.ttf", 20, NULL, ranges.Data);
  ImFont* font1 = io.Fonts->AddFontFromFileTTF("Fonts\\Poppins-Italic.otf", 20, &config);
  
  //Custom Font 2
  io.Fonts->AddFontFromFileTTF("Arial Unicode MS Font.ttf", 20, NULL, ranges.Data);
  ImFont* font2 = io.Fonts->AddFontFromFileTTF("Fonts\\Poppins-Bold.otf", 20, &config);

```

**Comment by ocornut** [comment]
>  But when i make this, application uses high ram and when \
> i add more than two font, application starts but imgui not showing

This means the texture is too big for your graphics API.

Dear ImGui way of prebaking glyphs ahead of time can't handle all languages very well yet. 
Adding every range is generally too much.
If you really need to do it you can reduce OversampleH/OversampleV values to 1.

But generally what you are trying to not do is not yet easily possible.

Read more
https://github.com/ocornut/imgui/blob/master/docs/FONTS.md

**Reply by ynd3fined** [reply]
Your method is works! Thanks for your support. 🤙
